posthog/api/decide.py

```diff
@@ -222,9 +222,13 @@ def get_decide(request: HttpRequest):
                 else False
             )
 
-            if settings.NEW_ANALYTICS_CAPTURE_TEAM_IDS and str(team.id) in settings.NEW_ANALYTICS_CAPTURE_TEAM_IDS:
-                if random() < settings.NEW_ANALYTICS_CAPTURE_SAMPLING_RATE:
-                    response["analytics"] = {"endpoint": settings.NEW_ANALYTICS_CAPTURE_ENDPOINT}
+            if str(team.id) not in settings.NEW_ANALYTICS_CAPTURE_EXCLUDED_TEAM_IDS:
+                if (
+                    "*" in settings.NEW_ANALYTICS_CAPTURE_TEAM_IDS
+                    or str(team.id) in settings.NEW_ANALYTICS_CAPTURE_TEAM_IDS
+                ):
+                    if random() < settings.NEW_ANALYTICS_CAPTURE_SAMPLING_RATE:
```

Comment:
I'm assuming you mean replicating plugin-server options such as `MAX_TEAM_ID_TO_BUFFER_ANONYMOUS_EVENTS_FOR`, where we ramped up that number progressively.

This is indeed a great approach when all of one team's data must be handled consistently, but we don't have such a concern with new capture (integration tests confirm that the shape in kafka is consistent with django). The main concern here is to control the ramp up in requests per seconds and active distinct IDs (for the memory usage of the overflow detection), and `random()` will give us a more predictable ramp up because these two dimensions will be linear with `NEW_ANALYTICS_CAPTURE_SAMPLING_RATE`. Ramping up the max_team_id instead would not give us the same linearity, as volumes/team are not consistent.

Reply:
👍  I trust you understand the details of your needs and potential failures better